File: sdks/python/cognitum/meta_llm/discovery.py
"""Discovery wire types: health, models, whoami (ADR-0024a §D1, §D2).

No service-owned OpenAPI contract exists yet (ADR-0024a §D9 gate #1), so
these stay intentionally permissive (``raw`` passthrough) rather than
pretending to be the eventual GA contract.
"""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any


@dataclass(frozen=True)
class MetaLlmHealth:
    """``health()`` response -- process-level only, never identity or readiness."""

    status: str
    version: str | None = None
    #: Unrecognized fields from the server response, preserved verbatim.
    raw: dict[str, Any] = field(default_factory=dict)


@dataclass(frozen=True)
class MetaLlmModelInfo:
    """A single entry from ``models()``.

    ``/v1/models`` may not list every accepted alias (ADR-0024a Context).
    """

    id: str
    object: str | None = None
    owned_by: str | None = None
    created: int | None = None
    raw: dict[str, Any] = field(default_factory=dict)


@dataclass(frozen=True)
class MetaLlmModelList:
    """``models()`` response."""

    models: list[MetaLlmModelInfo]
    object: str | None = None
    raw: dict[str, Any] = field(default_factory=dict)


@dataclass(frozen=True)
class MetaLlmWhoAmI:
    """``whoami()`` response -- authenticated account and credential type only."""

    account_id: str | None = None
    credential_type: str | None = None
    scopes: list[str] = field(default_factory=list)
    tenant_id: str | None = None
    raw: dict[str, Any] = field(default_factory=dict)


__all__ = ["MetaLlmHealth", "MetaLlmModelInfo", "MetaLlmModelList", "MetaLlmWhoAmI"]

File: sdks/node/src/meta-llm/discovery.ts
/**
 * Discovery wire types: health, models, whoami (ADR-0024a §D1, §D2).
 *
 * No service-owned OpenAPI contract exists yet (ADR-0024a §D9 gate #1), so
 * these stay intentionally permissive (`raw` passthrough) rather than
 * pretending to be the eventual GA contract.
 */

/** `health()` response — process-level only, never identity or readiness. */
export interface MetaLlmHealth {
  status: string;
  version?: string;
  /** Unrecognized fields from the server response, preserved verbatim. */
  raw?: Record<string, unknown>;
}

/** A single entry from `models()`. `/v1/models` may not list every accepted alias. */
export interface MetaLlmModelInfo {
  id: string;
  object?: string;
  ownedBy?: string;
  created?: number;
  raw?: Record<string, unknown>;
}

/** `models()` response. */
export interface MetaLlmModelList {
  object?: string;
  models: MetaLlmModelInfo[];
  raw?: Record<string, unknown>;
}

/** `whoami()` response — authenticated account and credential type only. */
export interface MetaLlmWhoAmI {
  accountId?: string;
  credentialType?: string;
  scopes?: string[];
  tenantId?: string;
  raw?: Record<string, unknown>;
}
